wikimedia/wikipedia-ios
# Image Recs - Bottom Modal Sheet

**Phabricator:** https://phabricator.wikimedia.org/T356818

### Notes
* This PR adds the bottom sheet component for Image recommendations
* On iPad and iPhone landscape, I'm considering an out-of-the-box approach for now until I get more input from design
* On the WKImageRecommendationBottomSheetView, in order to calculate the exclusion path to place the image on this almost columnar layout, it was necessary to calculate the size of the other view elements: the link button and the image view. I tried to keep the code as readable as possible, I'm open to suggestions here.
* I'm not loading the Image on the image view, I'm keeping the image view gray for now to help visualize the layout. Fetching, loading, and caching the image properly will be done in a separate ticket https://phabricator.wikimedia.org/T358933, since we need to build all the infrastructure for it

### Test Steps
1.  Launch the app on Staging or Experimental (I've been testing with English and Spanish)
2. Click the notifications button. You should see the modal sheet
3. Make sure you can swipe up to make it bigger, but it can't be dismissed
4. Test with larger fonts and other themes
5. Click "Not sure". Verify that the next suggestion loads correctly
6. Make sure you can scroll and reach the view article button on the article view
7. On iPad - it should no break, but the functionality needs to be adjusted
8. The yes, no, and link buttons are not on the scope of this task, so they have no action
